### ISSUE TITLE
Rows in destination are created and then deleted

### ISSUE THREAD

**Issue by stouch** [jitsucom/jitsu#1120]
## Summary

I setup the connection to my clickhouse database, with "segment" setup (because I wanted one table per event). 

I send the events using /track through HTTP API.

I then noticed that the rows are created in the correct clickhouse tables (and function are well applied) but then few minutes later the created rows were deleted from my clickhouse db .. After checking they well existed and then deleted.

Moreover, I noticed that the "tracks" table is filled with all the events that should appear in the tables, but of cours these events are not inserted in tables or deleted after insertion. Is this even possible than jitsu delete rows in destination ?

## System configuration and versions

k8s install of Jitsu v2.8.0

(https://github.com/stafftastic/jitsu-chart)

**Comment by vklimontovich** [comment]
Jitsu never deletes anything from the destination. You can see specific queries Jitsu runs on Clickhouse [here](https://docs.jitsu.com/destinations/warehouse/clickhouse) and [here](https://github.com/jitsucom/bulker/blob/main/.docs/db-feature-matrix.md#clickhouse). 

Most likely, it's on ClickHouse end. Are you using Clickhouse Cloud or self-hosted clickhouse?

**Comment by stouch** [comment]
OK sry yes, I understood what happened : 

https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replacingmergetree

>The engine differs from [MergeTree](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#table_engines-mergetree) in that it removes duplicate entries with the same [sorting key](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree) value (ORDER BY table section, not PRIMARY KEY).

>Data deduplication occurs only during a merge. Merging occurs in the background at an unknown time, so you can’t plan for it


For some reson, the sort key of my old jitsu tables  is : `eventn_ctx_event_id` 

But the problem is that now, this key became, `message_id`.

So my new rows had empty values in  `eventn_ctx_event_id` and they were randomly deleted on insert..

It's would be nice to have kind of retro-compatibility with the old sort key.

**Comment by stouch** [comment]
I confirm that my last comment explained what happened and it can affect your users.

I applied this function in my connection : 
```
export default async function(event, { log, fetch, props: config }) {
  return {
    ...event,
    properties: {
        ...event.properties,
        eventn_ctx_event_id: event.messageId,
        _timestamp: event.timestamp
    }
  }
}

```

which set the value of `message_id` in the old primary-key column (`eventn_ctx_event_id`). I also set the _timestamp column (which was the old timestamp column of my jitsu install).

Then I set the primary key of the connection as "eventn_ctx_event_id".

Now everything is working fine.
